### PR TITLE
tools: add debugging documentation

### PR DESCRIPTION
**Pull request by rubensfig**

### CODE REVIEW

File: tools/debugging.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+parent: Tools
+---
+
+# Debugging
+
+For debug purposes, refer to this page for instructions in enabling the debug options in BISDN Linux.
+
+## Debug instructions
+
+First check that all necessary services are up and running, with the following command. To ensure proper operation of BISDN Linux, baseboxd, ofdpa and ofagent must be running.
```
[the remark this answers is not in the document]
Comment (hilmarm) [reply tools/debugging.md:12]: Just for us stupid people so it's easy to see the names of the services. Also added ofdpa-grpc, although it very rarely is the issue for breaking anything.

File: tools/debugging.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+### glog
+#
+# log to stderr by default:
+GLOG_logtostderr=1
+
+# verbose log level:
+# GLOG_v=0
+```
+
+The relevant options for enabling debug logs in baseboxd is the `GLOG_v=0` value. By increasing the value in this configuration option, more detailed debug information will be outputted.
```
Comment (hilmarm) [inline tools/debugging.md:63]: What are the valid values? i.e. i=[0..N] where N is the largest integer

File: tools/debugging.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+Finer grained control of logging can be done by replacing the `GLOG_v=` configuration. When passing the baseboxd module names to `GLOG_vmodule`, which correspond to the class names, specific logging for that component will be activated with the specified logging level. The following command exemplifies how to configure this value.
+
+```
+GLOG_vmodule=<module name>=<log level>
+For example, setting logging to the main loop, the layer 3 and the controller classes,
+GLOG_vmodule=cnetlink=1,nl_l3=3,controller=2
```
Comment (hilmarm) [inline tools/debugging.md:70]: Nice example!

File: tools/debugging.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+# GLOG_v=0
+```
+
+The relevant options for enabling debug logs in baseboxd is the `GLOG_v=0` value. By increasing the value in this configuration option, more detailed debug information will be outputted. The accepted values are `GLOG_v=[1..5]`.
+
+When passing the baseboxd module names to `GLOG_vmodule`, which correspond to the class names, specific logging for that component will be activated with the specified logging level. The following command exemplifies how to configure this value.
+
+```
+GLOG_vmodule=<module name>=<log level>
+For example, setting logging to the main loop, the layer 3 and the controller classes,
```
Comment (jklare) [inline tools/debugging.md:81]: you can omit the the "For example...." here and describe above what exactly you are setting (including what the numbers chosen here mean)

File: tools/debugging.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+GLOG_vmodule=cnetlink=1,nl_l3=3,controller=2
+```
+
+### ofdpa debug files
+
+ofdpa is packaged with the configuration file `/etc/default/ofdpa`, visible below.
```
Comment (jklare) [inline tools/debugging.md:87]:
```suggestion
ofdpa ships with the default configuration file `/etc/default/ofdpa` shown below.
```

File: tools/debugging.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+#
+# debug level 0..4 (all components)
+# -d 2
+#
+# example:
+# OPTIONS="-c 1 -c 2 -d 2"
+OPTIONS=""
+```
+
+To activate the logging components inside ofdpa pass the `-c <component>` flag with the component, and to set the level for all components `-d <level>`.
```
Comment (jklare) [inline tools/debugging.md:112]:
```suggestion
To activate the logging for ofdpa, pass the component identifier (the numbers show in the default configuration file), together with the debug level you need to the `OPTIONS` key (e.g. `OPTIONS="-c 1 -c 5 -d 4"` to enable the maximum verbosity for the "API" and "DATAPATH" components).
```

File: tools/debugging.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+### ofagent debug files
+
+ofagent is packaged with the configuration file `/etc/default/ofagent`, visible below.
```
Comment (jklare) [inline tools/debugging.md:116]:
```suggestion
ofagent ships with the default configuration file `/etc/default/ofagent` shown below.
```

File: tools/debugging.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+# -t 10.10.10.10:6653
+#
+# agent level debugging 0..2
+# -a 2
+#
+# insert all options here:
+OPTIONS="-t 127.0.0.1:6653 -m"
+```
+
+Logging in ofagent can be controlled with the `-a <log level>` flag.
```
Comment (jklare) [inline tools/debugging.md:134]:
```suggestion
Logging in ofagent can be enabled by amending the `OPTIONS` key (please do not overwrite the default settings) with the `-a <log level>` flag (e.g. `OPTIONS="-t 127.0.0.1:6653 -m -a 2"` for maximum verbosity).
```

File: tools/debugging.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+parent: Tools
+---
+
+# Debugging
+
+For debug purposes, refer to this page for instructions on enabling the debug options in BISDN Linux.
+
+## Debug instructions
+
+To ensure proper operation of BISDN Linux, the service ``baseboxd``, ``ofdpa``,
```
Comment (ideaship) [inline tools/debugging.md:12]:
```suggestion
To ensure proper operation of BISDN Linux, the services ``baseboxd``, ``ofdpa``,
```

File: tools/debugging.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+```
+journalctl -u <service>
+```
+Check `man journalctl` for more information on the arguments.
+
+When trying to configure a feature, please remember to verify if the version
+and plaform being tested does support the feature. Make sure to check the list
+of known [limitations](https://docs.bisdn.de/limitations.html).
+
+To ensure that the link state itself is as expected, use `onlpdump -S` to get the state of all port (sample output below). If the port state and SFP information (not applicable to patch cables) does not correspond to the physical connections, debug the physical connection by adding another cable or try another module, if available.
```
Comment (ideaship) [inline tools/debugging.md:38]:
```suggestion
To ensure that the link state itself is as expected, use `onlpdump -S` to get the state of all ports (sample output below). If the port state and SFP information (not applicable to patch cables) does not correspond to the physical connections, debug the physical connection by adding another cable or try another module, if available.
```

**Pull request merged**